src/main/java/uk/gov/hmcts/reform/ccd/document/am/service/impl/DocumentManagementServiceImpl.java

```diff
@@ -41,17 +47,32 @@ public ResponseEntity getDocumentMetadata(UUID documentId) {
     }
 
     @Override
-    public String extractDocumentMetadata(StoredDocumentHalResource storedDocument) {
+    public String extractCaseIdFromMetadata(Object storedDocument) {
+        if (storedDocument instanceof StoredDocumentHalResource) {
+            Map<String,String> metadata = ((StoredDocumentHalResource) storedDocument).getMetadata();
+            return metadata.get("caseId");
```

Comment:
Few scenarios:
1. What if Metadata as field not available in storedDocument 
2. What if caseId key is not available in received map 
3. The caseId could be in uppercase, lower-case as well.
4. Case Id has to be only 16 character long and numeric number. 
- There should be some validations and appropriate error handling for each of such scenarios.

Reply:
What if Metadata as field not available in storedDocument
A. it will return caseid null but will not throw any exception.I tested it through postman.However, for subsequent call for ccd data store api ,I am checking caseid as null if it's  null then sending the response 404 with message "Case Id is missing in document meta data".

What if caseId key is not available in received map
A. Same as above.

The caseId could be in uppercase, lower-case as well.
A. I will check this.

Case Id has to be only 16 character long and numeric number
A.  I will check this as well.

Reply:
ok, In these cases we need to throw 403 saying the 'Insufficient permission on case document'. User doesn't know how this updated functionality works so giving information like metadata missing metadata is not available is not useful for it. Although we need to capture Error logs highlighting the missing metadata issue.